### PR TITLE
Events list widget single event markup

### PR DESCRIPTION
**Pull request by paulmskim**
**Ticket:** [TEC-3621]

This PR covers the FE templates for a single event in the events list widget. It includes the date tag, title, and date. The other pieces (venue, organizer, cost) are covered in another ticket. The styles are also covered in another ticket.

**Artifact:**

image

[TEC-3621]: https://moderntribe.atlassian.net/browse/TEC-3621

### CODE REVIEW

File: src/views/v2/widgets/events-list/event/title.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+<h3 class="tribe-events-widget-events-list__event-title tribe-common-h7">
+	<a
+		href="<?php echo esc_url( $event->permalink ); ?>"
+		title="<?php echo esc_attr( $event->title ); ?>"
+		rel="bookmark"
+		class="tribe-events-widget-events-list__event-title-link tribe-common-anchor-thin"
+	>
+		<?php
+		// phpcs:ignore
+		echo $event->title;
```
Comment (lucatume) [inline src/views/v2/widgets/events-list/event/title.php:28]: Should this be escaped? The code that produces that string is *not* escaping it: https://github.com/moderntribe/the-events-calendar/blob/5fc1b024de10e801f7799e250470f6035b97b273/src/Tribe/Models/Post_Types/Event.php#L237

File: src/views/v2/widgets/events-list/event/date.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+use Tribe__Date_Utils as Dates;
+
+$event_date_attr = $event->dates->start->format( Dates::DBDATEFORMAT );
+
+?>
+<div class="tribe-events-widget-events-list__event-datetime-wrapper tribe-common-b2 tribe-common-b3--min-medium">
+	<?php $this->template( 'widgets/events-list/event/date/featured', [ 'event' => $event ] ); ?>
+	<time class="tribe-events-widget-events-list__event-datetime" datetime="<?php echo esc_attr( $event_date_attr ); ?>">
+		<?php echo $event->schedule_details->value(); ?>
```
Comment (lucatume) [inline src/views/v2/widgets/events-list/event/date.php:26]: Drop a note here to indicate this is not escaped with the knowledge that what we get back is HTML built by us and already escaped.

File: src/views/v2/widgets/events-list/event/date-tag.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * past" in relation to the date they requested (or today's date).
+ */
+$display_date = empty( $is_past ) && ! empty( $request_date )
+	? max( $event->dates->start_display, $request_date )
+	: $event->dates->start_display;
+
+$event_month     = $display_date->format_i18n( 'M' );
+$event_day_num   = $display_date->format_i18n( 'j' );
+$event_date_attr = $display_date->format( Dates::DBDATEFORMAT );
+?>
```
Comment (lucatume) [inline src/views/v2/widgets/events-list/event/date-tag.php:32]:
```suggestion
?>

```

File: src/views/v2/widgets/events-list/event/cost.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @version TBD
+ *
+ * @var WP_Post            $event   The event post object with properties added by the `tribe_get_event` function.
+ * @var array<string,bool> $display Associative array of display settings for event meta.
+ *
+ * @see tribe_get_event() For the format of the event object.
+ */
+
+if ( empty( $event->cost ) || empty( $display['cost'] ) ) {
```
Comment (lucatume) [inline src/views/v2/widgets/events-list/event/cost.php:20]: Just checking, we're not showing the cost here if the cost is `0`, `'0'` or any other value that would be marked as empty, is this what we want here?

**Review by lucatume — CHANGES_REQUESTED** [review]

File: src/views/v2/widgets/events-list/event/title.php (hunk 1 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list/event/title.php:28]: I believe titles can contain HTML, which is why we don't escape it. This is how it is in the views as well.

File: src/views/v2/widgets/events-list/event/date-tag.php (hunk 3 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list/event/date-tag.php:32]: @lucatume not sure why this needs an extra space?

File: src/views/v2/widgets/events-list/event/cost.php (hunk 4 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list/event/cost.php:20]: Good question, this is how it is in list view currently. I'll check.

Comment (paulmskim) [reply src/views/v2/widgets/events-list/event/cost.php:20]: So for costs, if we enter `0` in the admin, it shows up as `Free` in the FE. If the field is left empty, it does not show.

File: src/views/v2/widgets/events-list/event/date.php (hunk 2 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list/event/date.php]: Done.

File: src/views/v2/widgets/events-list/event/date-tag.php (hunk 3 above)
Comment (lucatume) [reply src/views/v2/widgets/events-list/event/date-tag.php:32]: Formatting habit really.

**Review by lucatume — DISMISSED** [review]

File: src/views/v2/widgets/events-list/event/title.php (hunk 1 above)
Comment (Camwyn) [reply src/views/v2/widgets/events-list/event/title.php:28]: Late to the party, but `wp_kses_post` might be a reasonable alternative?

**Review by Camwyn — APPROVED** [review]

**Pull request merged**